File: java/com/epicdragonworld/gameserver/network/packets/sendable/MoveToLocation.java
package com.epicdragonworld.gameserver.network.packets.sendable;

import com.epicdragonworld.gameserver.model.WorldObject;
import com.epicdragonworld.gameserver.network.SendablePacket;

/**
 * @author Pantelis Andrianakis
 */
public class MoveToLocation extends SendablePacket
{
	public MoveToLocation(WorldObject object, float heading, int animState, int waterState)
	{
		// Send the data.
		writeShort(9); // Packet id.
		writeLong(object.getObjectId());
		writeFloat(object.getLocation().getX());
		writeFloat(object.getLocation().getY());
		writeFloat(object.getLocation().getZ());
		writeFloat(heading);
		writeShort(animState);
		writeByte(waterState);
	}
}

File: EpicDragonWorldServer/Network/Packets/Sendable/MoveToLocation.cs
/**
 * Author: Pantelis Andrianakis
 * Date: November 7th 2018
 */
class MoveToLocation : SendablePacket
{
    public MoveToLocation(WorldObject obj, float heading, int animState, int waterState)
    {
        // Send the data.
        WriteShort(9); // Packet id.
        WriteLong(obj.GetObjectId());
        WriteFloat(obj.GetLocation().GetX());
        WriteFloat(obj.GetLocation().GetY());
        WriteFloat(obj.GetLocation().GetZ());
        WriteFloat(heading);
        WriteShort(animState);
        WriteByte(waterState);
    }
}
